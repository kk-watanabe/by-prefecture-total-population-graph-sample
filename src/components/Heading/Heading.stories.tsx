import React from "react";

import { Heading } from "./Heading";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "Heading",
  component: Heading,
  argTypes: {
    tag: {
      options: ["h1", "h2", "h3", "h4", "h5", "h6"],
      control: { type: "radio" },
    },
    size: {
      options: ["4xl", "3xl", "2xl", "xl", "lg", "base", "sm", "xs"],
      control: { type: "radio" },
    },
  },
} as ComponentMeta<typeof Heading>;

const Template: ComponentStory<typeof Heading> = (args) => <Heading {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: "Heading label",
};

export const ChangeTag = Template.bind({});
ChangeTag.args = {
  label: "Heading label",
  tag: "h4",
};

export const ChangeSize = Template.bind({});
ChangeSize.args = {
  label: "Heading label",
  size: "2xl",
};
