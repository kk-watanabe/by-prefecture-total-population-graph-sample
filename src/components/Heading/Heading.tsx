import styles from "./Styles.module.css";

export type HeadingTagType = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
export type HeadingSizeType = "4xl" | "3xl" | "2xl" | "xl" | "lg" | "base" | "sm" | "xs";

export interface HeadingProps {
  tag?: HeadingTagType;
  size?: HeadingSizeType;
  label: string;
}

export const Heading = (props: HeadingProps) => {
  const Tag = props.tag !== undefined ? props.tag : "h1";
  const size = props.size !== undefined ? props.size : "4xl";
  const classes = [styles.heading, styles[`heading--size-${size}`]].join(" ");

  return <Tag className={classes}>{props.label}</Tag>;
};
