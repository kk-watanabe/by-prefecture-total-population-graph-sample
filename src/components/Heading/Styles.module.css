.heading {
  color: var(--color-gray800);
  font-weight: bold;
  line-height: 1.3;
}

.heading--size-4xl {
  font-size: var(--fs-4xl);
}
.heading--size-3xl {
  font-size: var(--fs-3xl);
}
.heading--size-2xl {
  font-size: var(--fs-2xl);
}
.heading--size-xl {
  font-size: var(--fs-xl);
}
.heading--size-lg {
  font-size: var(--fs-lg);
}
.heading--size-base {
  font-size: var(--fs-base);
}
.heading--size-sm {
  font-size: var(--fs-sm);
}
.heading--size-xs {
  font-size: var(--fs-xs);
}
