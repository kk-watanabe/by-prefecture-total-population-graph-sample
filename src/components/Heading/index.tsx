/* istanbul ignore next */
export * from "./Heading";
