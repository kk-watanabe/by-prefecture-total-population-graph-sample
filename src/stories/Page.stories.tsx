import React from "react";

import { Page } from "./Page";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "Example/Page",
  component: Page,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: "fullscreen",
  },
} as ComponentMeta<typeof Page>;

const Template: ComponentStory<typeof Page> = (args) => <Page {...args} />;

export const LoggedOut = Template.bind({});

export const LoggedIn = Template.bind({});
