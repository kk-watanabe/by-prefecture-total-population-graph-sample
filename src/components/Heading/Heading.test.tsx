import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

import * as stories from "./Heading.stories";

const { Default, ChangeTag, ChangeSize } = composeStories(stories);

describe("snapshots", () => {
  it("Readner Default components", () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("Readner ChangeTag components", () => {
    const { container } = render(<ChangeTag />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("Readner ChangeSize components", () => {
    const { container } = render(<ChangeSize />);

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("renders", () => {
  it("Readner Default components", () => {
    render(<Default />);

    expect(
      screen.getByRole("heading", {
        name: /Heading label/i,
        level: 1,
      })
    ).toBeInTheDocument();
  });

  it("Readner ChangeTag components", () => {
    render(<ChangeTag />);

    expect(
      screen.getByRole("heading", {
        name: /Heading label/i,
        level: 4,
      })
    ).toBeInTheDocument();
  });

  it("Readner ChangeSize components", () => {
    render(<ChangeSize />);

    expect(
      screen.getByRole("heading", {
        name: /Heading label/i,
        level: 1,
      })
    ).toBeInTheDocument();
  });
});
