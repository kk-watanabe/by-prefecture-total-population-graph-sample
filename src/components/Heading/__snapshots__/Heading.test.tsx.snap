// Vitest Snapshot v1

exports[`snapshots > Readner ChangeSize components 1`] = `
<h1
  class="_heading_e8u9f_1 _heading--size-2xl_e8u9f_13"
>
  Heading label
</h1>
`;

exports[`snapshots > Readner ChangeTag components 1`] = `
<h4
  class="_heading_e8u9f_1 _heading--size-4xl_e8u9f_7"
>
  Heading label
</h4>
`;

exports[`snapshots > Readner Default components 1`] = `
<h1
  class="_heading_e8u9f_1 _heading--size-4xl_e8u9f_7"
>
  Heading label
</h1>
`;
